src/main/java/com/google/devtools/build/lib/remote/blobstore/http/AbstractHttpHandler.java

```diff
@@ -34,6 +35,7 @@
 abstract class AbstractHttpHandler<T extends HttpObject> extends SimpleChannelInboundHandler<T>
     implements ChannelOutboundHandler {
 
+  private static final String USER_AGENT_VALUE = "bazel " + BlazeVersionInfo.instance().getReleaseName();
```

Comment:
We use `getVersion()` for the gRPC version (see TracingMetadataUtils) so I think it would make sense to also use it here for consistency. How about `BlazeVersionInfo.instance().isReleaseVersion() ? BlazeVersionInfo.instance().getVersion() : "dev"` or so. Please also then update the gRPC code to be consistent.

Reply:
Fair point. Do you think the logic you proposed should be a part of `getVersion()`?
Javadoc there says that it returns 'HEAD' in case of non-release bazel version, but code just returns an empty line. Maybe I should just fix that, cause I do not think it is a good idea to introduce another logic for retrieving bazel version.

Reply:
good point. please do so :-)

Reply:
done